src/AWSCredentials.jl

```diff
@@ -186,22 +190,34 @@ function localhost_is_ec2()
         return instance_uuid[1:3] == "EC2"
     end
 
+    # Note: try catch required for open calls on files of mode 400 (-r--------)
     # Note: This will not work on new m5 and c5 instances because they use a new hypervisor
     # stack and the kernel does not create files in sysfs
     hypervisor_uuid = "/sys/hypervisor/uuid"
-    if isfile(hypervisor_uuid) && _begins_with_ec2(hypervisor_uuid)
+    if isfile(hypervisor_uuid) && try isreadable(open(hypervisor_uuid, "r")) catch e; false; end
```

Comment:
Could we move these `try` blocks into their own lines?

```
hypervisor_isreadable = try ....
```

Makes the `if` statements a bit unsettling to read in their current form.

Reply:
Thanks for the feedback. I moved the two related file checks into a sub function.

In some ways I want Julia to have a `isreadable` on a filename that returns a boolean instead of needing to attempt to open it in read mode and catch the permission failure.  Checking the file stats or other methods in `Base.Filesystem` isn't any cleaner for seeing if the current user can read a file that it may or may not have permissions to read.

Accessing a read-only root filesystem is a pretty unique case of File I/O.  System level calls in Android probably fall in this same scenario, but it is unusual.